jlm/tooling/Command.cpp

```diff
@@ -270,6 +285,19 @@ LlcCommand::ToString(const RelocationModel & relocationModel)
 
 JlmOptCommand::~JlmOptCommand() = default;
 
+JlmOptCommand::JlmOptCommand(
+    std::string programName,
+    const jlm::tooling::JlmOptCommandLineOptions & commandLineOptions)
+    : ProgramName_(std::move(programName)),
+      CommandLineOptions_(std::move(commandLineOptions))
+{
+  for (auto optimizationId : CommandLineOptions_.GetOptimizationIds())
+  {
+    if (auto it = Optimizations_.find(optimizationId); it == Optimizations_.end())
```

Comment:
This initialization could have been done lazily in `GetOptimizations()`, is there a reason it is not?

Reply:
@haved The `Run()` method of the `JlmOptCommand` (or of any `Command` class) is marked as `const`. Thus, it is not permitted to modify the instance in this method. However, this method also invokes the `GetOptimizations()` method, which means that the `GetOptimization()` method also needs to be `const`. Thus, I could not have created and stored the optimizations in the `GetOptimization()` method. In the end, it does not make much of a difference whether the optimizations are initialized lazily or eagerly.

Reply:
Ah that's why. Resolving @phate